.github/workflows/go-CI.yml

```diff
@@ -0,0 +1,82 @@
+name: Go CI
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Set Up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.21.4' 
+
+      # - name: Install golangci-lint
+      #   run: go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+
+      # - name: Run golangci-lint
+      #   run: golangci-lint run
+
+      - name: Run Tests
```

Comment:
It's better to add cover flag to show test coverage ratio, also verbose to output more details of test runs